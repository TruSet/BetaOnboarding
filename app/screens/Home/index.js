import React, {Component} from 'react'
import injectSheet from 'react-jss'
import {Button} from '@material-ui/core'
import {WalkthroughModal} from '../../modal'
import config from '../../config'

// const mobileWidth = 1024

const styles = theme => ({
  root: {
    alignItems: 'center',
    flexDirection: 'column',
    backgroundColor: theme.palette.common.white,
    display: 'flex',
    justifyContent: 'space-around',
    height: 130,
    width: '100%',
  },
  thanks: {
    color: theme.palette.primary.main,
    fontWeight: 'bold',
  },
})

class home extends Component {
  state = {
    modal: false,
  }

  DEVCLEAR = () => {
    localStorage.clear()
    this.forceUpdate()
  }

  toggleModal = modal => () => this.setState({modal})

  render() {
    const {classes} = this.props
    const {modal} = this.state
    return (
      <div className={classes.root}>
        {modal && <WalkthroughModal close={this.toggleModal(false)}/>}
          <h3>TruSet Beta</h3>
          <div onClick={this.DEVCLEAR}>
            {!localStorage.getItem('applied')
              ? <Button variant='outlined' onClick={this.toggleModal(true)}>Apply Now</Button>
              : <span className={classes.thanks}>thanks for applying!</span>}
          </div>
          <h4>Begins {config.launchDate}</h4>
      </div>
    )
  }
}

export default injectSheet(styles)(home)

