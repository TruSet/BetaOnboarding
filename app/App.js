import {hot} from 'react-hot-loader'
import Home from './screens/Home'


export const Root = hot(module)(Home)
