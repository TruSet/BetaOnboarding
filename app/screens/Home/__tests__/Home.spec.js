import React from 'react'
import Home from '../index.js'
import {mountWithTheme} from '../../../App.spec'

describe('<Home />', () => {
  const component = mountWithTheme(<Home/>)
  expect(component).toMatchSnapshot()
  it('should render a modal when the button is clicked', () => {
    const modalRoot = global.document.createElement('div')
    modalRoot.setAttribute('id', 'modal-portal')
    const body = global.document.querySelector('body')
    body.appendChild(modalRoot)
    expect(component.find('div.walkthroughModal-background-0-1-3').exists()).toBe(false)
    component.setState({modal: true})
    expect(component.find('div.walkthroughModal-background-0-1-3').exists()).toBe(true)
    component.setState({modal: false})
  })
  it('should thank the user once application is done', () => {
    localStorage.setItem('applied', 'true')
    expect(component.find('.home-thanks-0-1-2')).toBeDefined()
  })
})
