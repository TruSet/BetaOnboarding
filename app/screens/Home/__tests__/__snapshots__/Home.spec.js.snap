// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<home
  classes={
    Object {
      "button": "home-button-0-1-2",
      "buttonText": "home-buttonText-0-1-3",
      "root": "home-root-0-1-1",
      "thanks": "home-thanks-0-1-4",
    }
  }
  theme={
    Object {
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "custom": Object {
        "defaultTitleHeight": 450,
      },
      "direction": "ltr",
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {},
      "palette": Object {
        "accent": Object {
          "blue": "rgb(61, 151, 211)",
          "red": "rgb(241, 90, 92)",
          "teal": "rgb(0, 176, 148)",
          "yellow": "rgb(254, 192, 23)",
        },
        "action": Object {
          "active": "rgba(0, 0, 0, 0.54)",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "disabledBackground": "rgba(0, 0, 0, 0.12)",
          "hover": "rgba(0, 0, 0, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(0, 0, 0, 0.14)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#fafafa",
          "paper": "#fff",
        },
        "common": Object {
          "black": "#000",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(0, 0, 0, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "primary": Object {
          "contrastText": "#fff",
          "dark": "rgb(41, 100, 46)",
          "light": "rgb(98, 166, 104)",
          "main": "#3b9043",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "rgb(23, 27, 30)",
          "light": "rgb(77, 82, 85)",
          "main": "#21272b",
        },
        "text": Object {
          "disabled": "rgba(0, 0, 0, 0.38)",
          "hint": "rgba(0, 0, 0, 0.38)",
          "primary": "#21272b",
          "secondary": "#6f7273",
        },
        "tonalOffset": 0.2,
        "type": "light",
      },
      "props": Object {},
      "shadows": Array [
        "none",
        "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
        "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
        "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
        "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
        "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
        "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
        "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
        "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
        "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
        "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
        "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
        "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
        "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
        "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
      ],
      "shape": Object {
        "borderRadius": 4,
      },
      "spacing": Object {
        "unit": 8,
      },
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "color": "#21272b",
          "fontFamily": "Muli, Roboto, sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 300,
          "lineHeight": "1.46429em",
        },
        "body2": Object {
          "color": "#6f7273",
          "fontFamily": "Muli, Helvetica, sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 300,
          "lineHeight": "1.71429em",
        },
        "button": Object {
          "color": "#21272b",
          "fontFamily": "Asap, Muli, Helvetica, sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 700,
          "textTransform": "uppercase",
        },
        "caption": Object {
          "color": "#6f7273",
          "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": "1.375em",
        },
        "display1": Object {
          "color": "#3b9043",
          "fontFamily": "Asap, Roboto, Helvetica, sans-serif",
          "fontSize": "2.125rem",
          "fontWeight": 700,
          "lineHeight": "1.20588em",
        },
        "display2": Object {
          "color": "#3b9043",
          "fontFamily": "Asap, Roboto, Helvetica, sans-serif",
          "fontSize": "2.8125rem",
          "fontStyle": "italic",
          "fontWeight": 700,
          "lineHeight": "1.13333em",
          "marginLeft": "-.02em",
        },
        "display3": Object {
          "color": "#6f7273",
          "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
          "fontSize": "3.5rem",
          "fontWeight": 400,
          "letterSpacing": "-.02em",
          "lineHeight": "1.30357em",
          "marginLeft": "-.02em",
        },
        "display4": Object {
          "color": "#6f7273",
          "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
          "fontSize": "7rem",
          "fontWeight": 300,
          "letterSpacing": "-.04em",
          "lineHeight": "1.14286em",
          "marginLeft": "-.04em",
        },
        "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
        "fontSize": 14,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "headline": Object {
          "color": "#21272b",
          "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": "1.35417em",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subheading": Object {
          "color": "#21272b",
          "fontFamily": "Glober,Muli,Asap,Roboto,Helvetica,Arial,sans-serif",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": "1.5em",
        },
        "title": Object {
          "color": "#21272b",
          "fontFamily": "Asap, Muli, Roboto, Helvetica, sans-serif",
          "fontSize": "1.3125rem",
          "fontWeight": 700,
          "lineHeight": "1.16667em",
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
>
  <div
    className="home-root-0-1-1"
  >
    <div>
      <WithStyles(Button)
        classes={
          Object {
            "label": "home-buttonText-0-1-3",
            "root": "home-button-0-1-2",
          }
        }
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        <Button
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-22",
              "contained": "MuiButton-contained-12",
              "containedPrimary": "MuiButton-containedPrimary-13",
              "containedSecondary": "MuiButton-containedSecondary-14",
              "disabled": "MuiButton-disabled-21",
              "extendedFab": "MuiButton-extendedFab-19",
              "fab": "MuiButton-fab-18",
              "flat": "MuiButton-flat-6",
              "flatPrimary": "MuiButton-flatPrimary-7",
              "flatSecondary": "MuiButton-flatSecondary-8",
              "focusVisible": "MuiButton-focusVisible-20",
              "fullWidth": "MuiButton-fullWidth-26",
              "label": "MuiButton-label-2 home-buttonText-0-1-3",
              "mini": "MuiButton-mini-23",
              "outlined": "MuiButton-outlined-9",
              "outlinedPrimary": "MuiButton-outlinedPrimary-10",
              "outlinedSecondary": "MuiButton-outlinedSecondary-11",
              "raised": "MuiButton-raised-15",
              "raisedPrimary": "MuiButton-raisedPrimary-16",
              "raisedSecondary": "MuiButton-raisedSecondary-17",
              "root": "MuiButton-root-1 home-button-0-1-2",
              "sizeLarge": "MuiButton-sizeLarge-25",
              "sizeSmall": "MuiButton-sizeSmall-24",
              "text": "MuiButton-text-3",
              "textPrimary": "MuiButton-textPrimary-4",
              "textSecondary": "MuiButton-textSecondary-5",
            }
          }
          color="primary"
          component="button"
          disableFocusRipple={false}
          disabled={false}
          fullWidth={false}
          mini={false}
          onClick={[Function]}
          size="medium"
          type="button"
          variant="contained"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-1 home-button-0-1-2 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-20"
            onClick={[Function]}
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-1 home-button-0-1-2 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-28",
                  "focusVisible": "MuiButtonBase-focusVisible-29",
                  "root": "MuiButtonBase-root-27",
                }
              }
              component="button"
              disableRipple={false}
              disableTouchRipple={false}
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="MuiButton-focusVisible-20"
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              <button
                className="MuiButtonBase-root-27 MuiButton-root-1 home-button-0-1-2 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-2 home-buttonText-0-1-3"
                >
                  Join the TruSet Token Beta Community
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-34",
                        "childLeaving": "MuiTouchRipple-childLeaving-35",
                        "childPulsate": "MuiTouchRipple-childPulsate-36",
                        "ripple": "MuiTouchRipple-ripple-31",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-33",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-32",
                        "root": "MuiTouchRipple-root-30",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-30"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-30"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
    </div>
  </div>
</home>
`;
